### ISSUE TITLE
Backup procedure

### ISSUE THREAD

**Issue by musicpanda** [thirtybees/thirtybees#1716]
Recently I encountered [an issue with the backup with Prestashop](https://github.com/PrestaShop/PrestaShop/issues/33818) that at least partially applies to Thirty Bees too. The problem is that the backup - as it is implemented in the backoffice - doesn't simply copy the content of the database. It also looks at the default values. If a field is empty and the default value is NULL it exports NULL instead of an empty string. 

I don't know Thirty Bees at the moment but in the newest Prestashops this causes serious problems. Besides that I don't think it is a good idea when a backup "improves" the data it backups.

The concerned code is in the function add() in the file /classes/PrestashopBackup.php.

The problematic code is 
`                        foreach ($row as $field => $value) {
                            $tmp = "'".pSQL($value, true)."',";
                            if ($tmp != "'',") {
                                $s .= $tmp;
                            } else {
                                foreach ($lines as $line) {
                                    if (strpos($line, '`'.$field.'`') !== false) {
                                        if (preg_match('/(.*NOT NULL.*)/Ui', $line)) {
                                            $s .= "'',";
                                        } else {
                                            $s .= 'NULL,';
                                        }
                                        break;
                                    }
                                }
                            }
                        }`

In my opinion it would be better to have it like this:
`			foreach ($row as $field => $value) 
			{	$tmp = "'".pSQL($value, true)."',";
				if (is_null($value)) 
				{ 
					$str .= 'NULL,';
				}
				else
				{
					$str.= $tmp;
				}
			}`

**Comment by getdatakick** [comment]
I agree, backup system should indeed backup data exactly as they are in database, and not perform any unnecessary optimization and transformations.
